pkg/statistics/handle/cache/statscache.go

```diff
@@ -186,12 +193,21 @@ func (s *StatsCacheImpl) Update(ctx context.Context, is infoschema.InfoSchema, t
 		}
 		tableInfo := table.Meta()
 		// If the table is not updated, we can skip it.
-		if oldTbl, ok := s.Get(physicalID); ok &&
-			oldTbl.Version >= version &&
+
+		oldTbl, ok := s.Get(physicalID)
+		if ok && oldTbl.Version >= version &&
 			tableInfo.UpdateTS == oldTbl.TblInfoUpdateTS {
 			continue
 		}
-		tbl, err := s.statsHandle.TableStatsFromStorage(
+		var tbl *statistics.Table
+		// If the column/index stats has not been updated, we can reuse the old table stats.
+		// Only need to update the count and modify count.
+		if ok && latestHistUpdateVersion > 0 && oldTbl.LastStatsUpdateVersion >= latestHistUpdateVersion {
+			tbl = oldTbl.Copy()
+			// count and modify count is updated in finalProcess
+			goto finalProcess
```

Comment:
I think better to avoid `goto` here. I guess we can use normal if else to workaround?